### PR TITLE
Make basstreble stateless

### PR DESCRIPTION
**Pull request by pietro68**
Resolves: #3626 

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/effects/BassTreble.cpp [hunk 1]
```diff
@@ -110,78 +200,94 @@ auto EffectBassTreble::RealtimeSupport() const -> RealtimeSince
...
       return 0;
-   return InstanceProcess(settings, mSlaves[group], inbuf, outbuf, numSamples);
+   return InstanceProcess(settings, mSlaves[group].mState, inbuf, outbuf, numSamples);
 }
 
-bool EffectBassTreble::CheckWhetherSkipEffect(const EffectSettings &) const
+bool EffectBassTreble::CheckWhetherSkipEffect(const EffectSettings& settings) const
 {
-   return (mBass == 0.0 && mTreble == 0.0 && mGain == 0.0);
+   auto& ms = GetSettings(settings);;
```
Comment (Paul-Licameli) [inline src/effects/BassTreble.cpp:267]: extra ';'

File: src/effects/BassTreble.cpp [hunk 2]
```diff
@@ -61,20 +53,118 @@ const ComponentInterfaceSymbol EffectBassTreble::Symbol
...
+   bool RealtimeFinalize(EffectSettings& settings) noexcept override;
+
+   size_t RealtimeProcess(size_t group, EffectSettings& settings,
+      const float* const* inbuf, float* const* outbuf, size_t numSamples)
+      override;
+
+   unsigned GetAudioInCount() const override;
+   unsigned GetAudioOutCount() const override;
+
+   void InstanceInit(EffectSettings& settings, EffectBassTrebleState& data, float sampleRate);
```
[the remark this answers is not in the document]
Comment (Paul-Licameli) [reply src/effects/BassTreble.cpp:141]: The three functions that follow can also compile as `static`.

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]
Minor suggestions.  This looks good.

Now, I think we should reenable BassTreble, Phaser, and Wahwah in the side bar.

So add a commit to do that.  Then let QA test those three effects before this is merged.

Echo is another effect that could easily join those three.  The transformation of a PerTrackEffect for realtime support should be a very familiar exercise now.  I suggest that as another near term task.

File: src/effects/BassTreble.cpp (hunk 1 above)
Comment (pietro68) [reply src/effects/BassTreble.cpp]: ok, done

File: src/effects/BassTreble.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (pietro68) [reply src/effects/BassTreble.cpp]: ok, done

**Review by Paul-Licameli — APPROVED** [review]

**Pull request merged**